src/controllers/userController.js

```diff
@@ -0,0 +1,70 @@
+import userService from '../services/userService';
+import { jwtSignUser } from '../utils/index';
+import { comparePassword } from '../helpers/comparePassword';
+import { logins } from '../models';
+/**
+ * @class UsersController
+ */
+class UserController {
+  /** Login User
+   * @static
+   * @param {object} req - HTTP Request
+   * @param {object} res - HTTP Response
+   * @returns {string} loginUsers
+   */
+  static async loginAUser(req, res) {
+    const { email, password } = req.body;
+    try {
+      const loggedUser = await userService.loginAUser(email);
+      if (!loggedUser) {
+        return res.status(404).json({
+          error: 'Invalid login details',
+        });
+      }
+      const userDetail = {
+        id: loggedUser.dataValues.id,
+        email: loggedUser.dataValues.email,
+      };
+      if (loggedUser) {
+        const validPassword = await comparePassword(
+          loggedUser.password,
+          password
+        );
+        if (validPassword) {
+          const lastLogin = await userService.findLastLogin(
+            loggedUser.dataValues.email
+          );
+          if (!lastLogin) {
+            const saveUserLoggedTime = {
+              email: loggedUser.dataValues.email,
+              password: loggedUser.password,
+            };
+            await userService.addLoggedInUser(saveUserLoggedTime);
+          } else {
+            await logins.update({ lastLogin: new Date(), }, {
+              where: {
+                email: loggedUser.email
+              }
+            });
+          }
+          const token = await jwtSignUser(userDetail);
```

Comment:
Why do you generate token after you have updated the login for this user in the database. If an error occured during token generation or delivery, won't the system think that this user has successfully logged in?